// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/TextPost/TextPost.js
import { jsx } from 'react/jsx-runtime';
import { tokens } from '../../../constants/tokens/index.js';
import Container from '../Container/Container.js';
import Markdown, { MarkdownTheme } from '../Markdown/Markdown.js';
import VerticalSpacing from '../VerticalSpacing/VerticalSpacing.js';
import markdownStyles from '../Markdown/Markdown.css.js';

function TextPost(props) {
    const { children , ariaLabelledBy , linkComponent , themeColor =tokens.colorCoreBrand , testHook ,  } = props;
    if (!children) return null;
    return jsx(Container, Object.assign({
        maxWidth: "sizeRoot",
        testHook: testHook
    }, {
        children: jsx(VerticalSpacing, Object.assign({
            size: "large8X"
        }, {
            children: jsx(MarkdownTheme, Object.assign({
                color: themeColor
            }, {
                children: jsx(Markdown, Object.assign({
                    as: "article",
                    className: markdownStyles.textPost,
                    aria: ariaLabelledBy ? {
                        'aria-labelledby': ariaLabelledBy
                    } : undefined,
                    linkComponent: linkComponent,
                    testHook: testHook && `${testHook}-article`
                }, {
                    children: children
                }), void 0)
            }), void 0)
        }), void 0)
    }), void 0);
}

export { TextPost as default };
