// @getatomi/neon@33.2.2-PROD-4806.1/dist/constants/sizes.js
import keyMirror from 'keymirror';

// keep sizes ordered from low to high (not alphabetically) since they are used in docs
const sizes = keyMirror({
    none: null,
    small2X: null,
    small1X: null,
    small: null,
    root: null,
    large: null,
    large1X: null,
    large2X: null,
    large3X: null,
    large4X: null,
    large5X: null,
    large6X: null,
    large7X: null,
    large8X: null,
    large9X: null
});
keyMirror({
    [sizes.small1X]: null,
    [sizes.small]: null,
    [sizes.root]: null,
    [sizes.large]: null
});
keyMirror({
    [sizes.small]: null
});
keyMirror({
    sizeSmall2X: null,
    sizeSmall1X: null,
    sizeSmall: null,
    sizeRoot: null,
    sizeLarge: null,
    sizeLarge1X: null,
    sizeFull: null
});
const dateTimeSizes = keyMirror({
    [sizes.small]: null
});
keyMirror({
    [sizes.small]: null
});
const headingSizes = keyMirror({
    [sizes.small1X]: null,
    [sizes.small]: null,
    medium: null,
    [sizes.large]: null,
    [sizes.large1X]: null,
    [sizes.large2X]: null
});
keyMirror({
    small3X: null,
    [sizes.small2X]: null,
    [sizes.small1X]: null,
    [sizes.small]: null,
    [sizes.root]: null,
    [sizes.large]: null,
    [sizes.large1X]: null,
    [sizes.large2X]: null,
    [sizes.large3X]: null,
    [sizes.large4X]: null
});
const progressCircleSizes = keyMirror({
    [sizes.small1X]: null,
    [sizes.small]: null,
    [sizes.root]: null,
    [sizes.large]: null,
    [sizes.large1X]: null
});
keyMirror({
    [sizes.small]: null
});
const imageSizes = keyMirror({
    [sizes.small1X]: null,
    [sizes.small]: null,
    [sizes.root]: null,
    [sizes.large]: null
});
keyMirror({
    [sizes.large]: null
});
keyMirror({
    [sizes.small]: null
});
const spinnerSizes = keyMirror({
    [sizes.small1X]: null,
    [sizes.small]: null
});
keyMirror({
    [sizes.large]: null
});
keyMirror({
    [sizes.large]: null
});
keyMirror({
    [sizes.large]: null
});

export { dateTimeSizes, sizes as default, headingSizes, imageSizes, progressCircleSizes, spinnerSizes };
