// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/Dialog/DialogBase.js
import { jsx } from 'react/jsx-runtime';
import ReactModal from 'react-modal';
import cx from 'classnames';
import styles from './DialogBase.css.js';

function DialogBase(props) {
    const { appElement , ariaLabel , children , className , headingId , isOpen , onClose , overlayClassName , ...otherProps } = props;
    ReactModal.setAppElement(appElement);
    return jsx(ReactModal, Object.assign({
        // @ts-ignore react-modal accepts any aria-* property
        aria: {
            labelledby: headingId,
            label: ariaLabel
        },
        htmlOpenClassName: styles.isOpen,
        className: cx(styles.root, className),
        isOpen: isOpen,
        // TS: force assertion from a Button event to a MouseEvent
        onRequestClose: onClose,
        overlayClassName: cx(styles.overlay, overlayClassName),
        role: "dialog",
        shouldCloseOnEsc: true,
        shouldReturnFocusAfterClose: true
    }, otherProps, {
        children: children
    }), void 0);
}

export { DialogBase as default };
