// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/TimePicker/TimePicker.js
import { jsx } from 'react/jsx-runtime';
import _ from 'lodash';
import moment from 'moment';
import SelectField from '../SelectField/SelectField.js';

const labelFormat = 'h a';
function get12hClock(inputFormat, range) {
    const min = range && moment(range.min, inputFormat);
    const max = range && moment(range.max, inputFormat);
    const current = moment().set({
        minute: 0,
        second: 0,
        millisecond: 0
    });
    return _.range(0, 24).reduce((acc, hour)=>{
        current.set({
            hour
        });
        if (!range || min && max && current.isBetween(min, max, 'minute', '[]')) {
            return acc.concat({
                label: current.format(labelFormat),
                value: current.format(inputFormat)
            });
        }
        return acc;
    }, []);
}
function TimePicker(props) {
    const { inputFormat , range , ...selectProps } = props;
    return jsx(SelectField, Object.assign({}, selectProps, {
        options: get12hClock(inputFormat, range)
    }), void 0);
}

export { TimePicker as default };
