// @getatomi/neon@33.2.2-PROD-4806.1/dist/constants/tokens/boxShadows.js
import { colors } from './colors.js';

const boxShadowWidthRoot = '0 0 0 1px';
const boxShadowWidthLarge = '0 0 0 2px';
// More info about 8-Digit Hex Color Code: https://css-tricks.com/8-digit-hex-codes/
const boxShadowColorAlpha10 = '1A'; // 10%
const boxShadowColorAlpha80 = 'CC'; // 80%
const boxShadows = Object.freeze({
    boxShadow: `0 10px 20px 0 ${colors.colorCoreBlack}${boxShadowColorAlpha10}`,
    boxShadowHigh: `0 0 20px 0 ${colors.colorFiftyShadesLighter}`,
    boxShadowBorder: `${boxShadowWidthRoot} ${colors.colorFiftyShadesLighter}`,
    boxShadowBorderFocused: `${boxShadowWidthRoot} ${colors.colorCoreBrand}`,
    boxShadowBorderError: `${boxShadowWidthRoot} ${colors.colorStatusErrorRoot}`,
    boxShadowBorderErrorFocused: `${boxShadowWidthRoot} ${colors.colorStatusErrorDark}`,
    boxShadowBorderWarning: `${boxShadowWidthRoot} ${colors.colorStatusWarningDark}`,
    boxShadowBorderWarningFocused: `${boxShadowWidthRoot} ${colors.colorStatusWarningDarker}`,
    boxShadowBorderInset: `inset ${boxShadowWidthRoot} ${colors.colorFiftyShadesLighter}`,
    boxShadowBorderInsetDisabled: `inset ${boxShadowWidthRoot} ${colors.colorFiftyShadesLightest}`,
    boxShadowBorderInsetFocused: `inset ${boxShadowWidthRoot} ${colors.colorInteractiveLink}`,
    boxShadowBorderInsetSelected: `inset ${boxShadowWidthLarge} ${colors.colorCoreBrand}`,
    boxShadowBorderInsetError: `inset ${boxShadowWidthRoot} ${colors.colorStatusErrorRoot}`,
    boxShadowBorderInsetErrorFocused: `inset ${boxShadowWidthRoot} ${colors.colorStatusErrorDark}`,
    boxShadowBorderInsetWarning: `inset ${boxShadowWidthRoot} ${colors.colorStatusWarningDark}`,
    boxShadowBorderInsetWarningFocused: `inset ${boxShadowWidthRoot} ${colors.colorStatusWarningDarker}`,
    boxShadowBorderInsetLargeCorrect: `inset ${boxShadowWidthLarge} ${colors.colorGreenMonsterDark}`,
    boxShadowBorderInsetLargeError: `inset ${boxShadowWidthLarge} ${colors.colorStatusErrorRoot}`,
    boxShadowBorderInsetLargeWarning: `inset ${boxShadowWidthLarge} ${colors.colorStatusWarningDark}`,
    boxShadowCard: `2px 2px 10px 0 ${colors.colorFiftyShadesLighter}`,
    boxShadowCardHover: `2px 2px 10px 0 ${colors.colorFiftyShadesLight}${boxShadowColorAlpha80}`
});

export { boxShadows };
