// @getatomi/neon@33.2.2-PROD-4806.1/dist/constants/tokens/breakpoints.js
import { pxToEm } from '../../utils/pxToRelativeUnit.js';

const breakpointWidths = {
    small1X: 360,
    small: 480,
    medium: 768,
    large: 992,
    large1X: 1200
};
const breakpoints = Object.freeze({
    breakpointLarge: pxToEm(breakpointWidths.large),
    breakpointLargeFrom: pxToEm(breakpointWidths.large + 1),
    breakpointLarge1X: pxToEm(breakpointWidths.large1X),
    breakpointLarge1XFrom: pxToEm(breakpointWidths.large1X + 1),
    breakpointMedium: pxToEm(breakpointWidths.medium),
    breakpointMediumFrom: pxToEm(breakpointWidths.medium + 1),
    breakpointSmall1X: pxToEm(breakpointWidths.small1X),
    breakpointSmall1XFrom: pxToEm(breakpointWidths.small1X + 1),
    breakpointSmall: pxToEm(breakpointWidths.small),
    breakpointSmallFrom: pxToEm(breakpointWidths.small + 1)
});
// Named breakpoints used by the ThemeProvider for responsive system props
const themeBreakpoints = {
    mobile: breakpoints.breakpointSmall,
    tablet: breakpoints.breakpointMedium,
    desktop: breakpoints.breakpointLarge,
    wide: breakpoints.breakpointLarge1X
};

export { breakpoints, themeBreakpoints };
