// @getatomi/neon@33.2.2-PROD-4806.1/dist/constants/tokens/colors.js
import keyMirror from 'keymirror';

const shared = {
    colorCoreBlack: '#303338',
    colorCoreBrand: '#177cf9',
    colorInteractiveLink: '#066fef'
};
keyMirror({
    colorArcticIce: null,
    colorAtomiBlues: null,
    colorAwkwardOrchid: null,
    colorBlueSkies: null,
    colorFairyFloss: null,
    colorFiftyShades: null,
    colorGreenMonster: null,
    colorMadMagenta: null,
    colorMatchaLatte: null,
    colorOceanSwim: null,
    colorPurpleRain: null,
    colorStatusError: null,
    colorStatusSuccess: null,
    colorStatusWarning: null,
    colorStrawberryFields: null,
    colorSunnyArvo: null,
    colorTangerineTease: null
});
const colors = Object.freeze({
    // `null` values means product design are yet to define the colour
    // Core
    colorCoreBlack: shared.colorCoreBlack,
    colorCoreBrand: shared.colorCoreBrand,
    colorCoreText: shared.colorCoreBlack,
    colorCoreWhite: '#fff',
    // Interactive
    colorInteractiveFocusRing: shared.colorInteractiveLink,
    colorInteractiveLink: shared.colorInteractiveLink,
    colorInteractiveLinkHover: shared.colorCoreBlack,
    // Arctic ice
    colorArcticIceDark: '#007ea8',
    colorArcticIceDarker: '#006486',
    colorArcticIceDarkest: '#005172',
    colorArcticIceLight: null,
    colorArcticIceLighter: '#c1e9f4',
    colorArcticIceLightest: '#e7f6fa',
    colorArcticIceRoot: '#65cae4',
    // Atomi blues
    colorAtomiBluesDark: shared.colorInteractiveLink,
    colorAtomiBluesDarker: '#1a61ad',
    colorAtomiBluesDarkest: '#344563',
    colorAtomiBluesLight: '#9bc5f8',
    colorAtomiBluesLighter: '#d9e9fc',
    colorAtomiBluesLightest: '#f3f8ff',
    colorAtomiBluesRoot: shared.colorCoreBrand,
    // Awkward orchid
    colorAwkwardOrchidDark: '#935cbc',
    colorAwkwardOrchidDarker: '#804aad',
    colorAwkwardOrchidDarkest: '#6c3997',
    colorAwkwardOrchidLight: null,
    colorAwkwardOrchidLighter: '#e0caf0',
    colorAwkwardOrchidLightest: '#f3ebf9',
    colorAwkwardOrchidRoot: '#b27cda',
    // Blue skies
    colorBlueSkiesDark: null,
    colorBlueSkiesDarker: '#007aa3',
    colorBlueSkiesDarkest: '#006080',
    colorBlueSkiesLight: '#66caeb',
    colorBlueSkiesLighter: '#ceedf8',
    colorBlueSkiesLightest: '#e6f6fb',
    colorBlueSkiesRoot: '#00a7dd',
    // Fairy floss
    colorFairyFlossDark: null,
    colorFairyFlossDarker: '#d33c5f',
    colorFairyFlossDarkest: '#a8304c',
    colorFairyFlossLight: null,
    colorFairyFlossLighter: '#f6cfda',
    colorFairyFlossLightest: '#fbecf0',
    colorFairyFlossRoot: '#e988a3',
    // Fifty shades
    colorFiftyShadesDark: '#6e7680',
    colorFiftyShadesDarker: '#5b636c',
    colorFiftyShadesDarkest: shared.colorCoreBlack,
    colorFiftyShadesLight: '#bfc4cb',
    colorFiftyShadesLighter: '#e1e4e7',
    colorFiftyShadesLightest: '#f8fafb',
    colorFiftyShadesRoot: '#9ca3ac',
    // Green monster
    colorGreenMonsterDark: '#00a87e',
    colorGreenMonsterDarker: '#007a5c',
    colorGreenMonsterDarkest: '#00674a',
    colorGreenMonsterLight: null,
    colorGreenMonsterLighter: '#99ead6',
    colorGreenMonsterLightest: '#d8f7ef',
    colorGreenMonsterRoot: '#00cc99',
    // Mad magenta
    colorMadMagentaDark: null,
    colorMadMagentaDarker: '#a20059',
    colorMadMagentaDarkest: '#810047',
    colorMadMagentaLight: null,
    colorMadMagentaLighter: '#e6afcd',
    colorMadMagentaLightest: '#f5e1ec',
    colorMadMagentaRoot: '#c13883',
    // Matcha latte
    colorMatchaLatteDark: null,
    colorMatchaLatteDarker: '#88b05f',
    colorMatchaLatteDarkest: '#537f24',
    colorMatchaLatteLight: null,
    colorMatchaLatteLighter: '#e3f3d1',
    colorMatchaLatteLightest: '#f4faed',
    colorMatchaLatteRoot: '#bae191',
    // Ocean swim
    colorOceanSwimDark: null,
    colorOceanSwimDarker: '#007c85',
    colorOceanSwimDarkest: '#00636a',
    colorOceanSwimLight: null,
    colorOceanSwimLighter: '#99e9e7',
    colorOceanSwimLightest: '#d9f6f6',
    colorOceanSwimRoot: '#00c9c3',
    // Purple rain
    colorPurpleRainDark: null,
    colorPurpleRainDarker: '#48418b',
    colorPurpleRainDarkest: '#39346f',
    colorPurpleRainLight: null,
    colorPurpleRainLighter: '#c5badd',
    colorPurpleRainLightest: '#e9e5f2',
    colorPurpleRainRoot: '#6f54aa',
    // Status error
    colorStatusErrorDark: '#dd002f',
    colorStatusErrorDarker: '#c82626',
    colorStatusErrorDarkest: '#a01e1e',
    colorStatusErrorLight: null,
    colorStatusErrorLighter: '#ffbdbf',
    colorStatusErrorLightest: '#ffe5e6',
    colorStatusErrorRoot: '#ff5a5f',
    // Status success
    colorStatusSuccessDark: '#00A87E',
    colorStatusSuccessDarker: '#007A5C',
    colorStatusSuccessDarkest: '#00674A',
    colorStatusSuccessLight: null,
    colorStatusSuccessLighter: '#99EAD6',
    colorStatusSuccessLightest: '#D8F7EF',
    colorStatusSuccessRoot: '#00CC99',
    // Status warning
    colorStatusWarningDark: '#f86530',
    colorStatusWarningDarker: '#d43e08',
    colorStatusWarningDarkest: '#9d2e06',
    colorStatusWarningLight: null,
    colorStatusWarningLighter: '#fdd9cd',
    colorStatusWarningLightest: '#feece6',
    colorStatusWarningRoot: '#f9845b',
    // Strawberry fields
    colorStrawberryFieldsDark: '#dd002f',
    colorStrawberryFieldsDarker: '#c82626',
    colorStrawberryFieldsDarkest: '#a01e1e',
    colorStrawberryFieldsLight: null,
    colorStrawberryFieldsLighter: '#ffbdbf',
    colorStrawberryFieldsLightest: '#ffe5e6',
    colorStrawberryFieldsRoot: '#ff5a5f',
    // Sunny arvo
    colorSunnyArvoDark: '#fcb95b',
    colorSunnyArvoDarker: '#de935d',
    colorSunnyArvoDarkest: '#a76553',
    colorSunnyArvoLight: null,
    colorSunnyArvoLighter: '#feefc5',
    colorSunnyArvoLightest: '#fef9e9',
    colorSunnyArvoRoot: '#fed86f',
    // Tangerine tease
    colorTangerineTeaseDark: '#d43e08',
    colorTangerineTeaseDarker: '#bd4428',
    colorTangerineTeaseDarkest: '#622214',
    colorTangerineTeaseLight: null,
    colorTangerineTeaseLighter: '#fdd9cd',
    colorTangerineTeaseLightest: '#feece6',
    colorTangerineTeaseRoot: '#f9845b',
    // Transparency
    colorTransparent: 'transparent',
    colorWhiteTransparent20: '#FFFFFF20',
    colorWhiteTransparent40: '#FFFFFF40',
    colorWhiteTransparent70: '#FFFFFF70'
});

export { colors };
