// @getatomi/neon@33.2.2-PROD-4806.1/dist/constants/tokens/spacing.js
import { pxToRem } from '../../utils/pxToRelativeUnit.js';
import rootPixelSizes from '../rootPixelSizes.js';

const { rootPixelSizeNeon: __16  } = rootPixelSizes;
const spacing = Object.freeze({
    spacingNone: pxToRem(0),
    spacingSmall4X: pxToRem(1),
    spacingSmall3X: pxToRem(__16 - 14),
    spacingSmall2X: pxToRem(__16 - 12),
    spacingSmall1X: pxToRem(__16 - 8),
    spacingSmall: pxToRem(__16 - 4),
    spacingRoot: pxToRem(__16),
    spacingLarge: pxToRem(__16 + 4),
    spacingLarge1X: pxToRem(__16 + 8),
    spacingLarge2X: pxToRem(__16 + 12),
    spacingLarge3X: pxToRem(__16 + 16),
    spacingLarge4X: pxToRem(__16 + 20),
    spacingLarge5X: pxToRem(__16 + 24),
    spacingLarge6X: pxToRem(__16 + 32),
    spacingLarge7X: pxToRem(__16 + 40),
    spacingLarge8X: pxToRem(__16 + 56),
    spacingLarge9X: pxToRem(__16 + 80),
    // negative spacing tokens
    spacingNegativeSmall4X: pxToRem(-1),
    spacingNegativeSmall3X: pxToRem(-__16 + 14),
    spacingNegativeSmall2X: pxToRem(-__16 + 12),
    spacingNegativeSmall1X: pxToRem(-__16 + 8),
    spacingNegativeSmall: pxToRem(-__16 + 4),
    spacingNegativeRoot: pxToRem(-__16),
    spacingNegativeLarge: pxToRem(-__16 - 4),
    spacingNegativeLarge1X: pxToRem(-__16 - 8),
    spacingNegativeLarge2X: pxToRem(-__16 - 12),
    spacingNegativeLarge3X: pxToRem(-__16 - 16),
    spacingNegativeLarge4X: pxToRem(-__16 - 20),
    spacingNegativeLarge5X: pxToRem(-__16 - 24),
    spacingNegativeLarge6X: pxToRem(-__16 - 32),
    spacingNegativeLarge7X: pxToRem(-__16 - 40),
    spacingNegativeLarge8X: pxToRem(-__16 - 56),
    spacingNegativeLarge9X: pxToRem(-__16 - 80)
});

export { spacing };
