// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/HelpInfo/HelpInfo.js
import { jsxs, jsx } from 'react/jsx-runtime';
import cx from 'classnames';
import keyMirror from 'keymirror';
import HideVisually from '../HideVisually/HideVisually.js';
import Icon from '../Icon/Icon.js';
import Text from '../Text/Text.js';
import styles from './HelpInfo.css.js';

keyMirror({
    leftAligned: null
});
function HelpInfo(props) {
    const { assistiveText ='Help info: ' , children , testHook , variant  } = props;
    return jsxs("small", Object.assign({
        className: cx(styles.root, {
            [String(variant && styles[variant])]: variant
        }),
        "data-test": testHook
    }, {
        children: [
            jsx(Icon, {
                className: styles.icon,
                size: "sizeIconSmall",
                name: "circle-help"
            }, void 0),
            jsxs("span", Object.assign({
                className: styles.info
            }, {
                children: [
                    jsx(HideVisually, {
                        children: assistiveText
                    }, void 0),
                    jsx(Text, Object.assign({
                        as: "span",
                        variant: "bodySmall1X",
                        color: "colorFiftyShadesDark"
                    }, {
                        children: children
                    }), void 0)
                ]
            }), void 0)
        ]
    }), void 0);
}

export { HelpInfo as default };
