// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/Modal/ModalFooter.js
import { jsx } from 'react/jsx-runtime';
import Box from '../Box/Box.js';
import { defaultSize } from './Modal.js';

const sharedStyles = {
    flexShrink: 0,
    marginTop: 'auto'
};
const mediumStyles = {
    ...sharedStyles,
    borderTopColor: 'colorFiftyShadesLighter',
    borderTopStyle: 'solid',
    borderTopWidth: 'borderWidthRoot',
    marginTop: 'auto',
    paddingBlock: {
        base: 'spacingRoot',
        tablet: 'spacingLarge1X'
    },
    paddingInline: {
        base: 'spacingRoot',
        tablet: 'spacingLarge3X'
    }
};
const smallStyles = {
    ...sharedStyles,
    paddingBottom: {
        base: 'spacingLarge1X',
        tablet: 'spacingLarge3X'
    },
    paddingInline: {
        base: 'spacingLarge1X',
        tablet: 'spacingLarge3X'
    }
};
const sizeStylesMapping = {
    medium: mediumStyles,
    small: smallStyles
};
function ModalFooter(props) {
    const { children , size =defaultSize  } = props;
    return jsx(Box, Object.assign({}, sizeStylesMapping[size], {
        children: children
    }), void 0);
}

export { ModalFooter as default };
