// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/ButtonAdd/ButtonAdd.js
import { jsx } from 'react/jsx-runtime';
import { tokens } from '../../../constants/tokens/index.js';
import Button, { ButtonLoader } from '../Button/Button.js';
import IconButton from '../IconButton/IconButton.js';
import { useMedia } from '../Media/Media.js';
import Skeleton from '../Skeleton/Skeleton.js';

function ButtonAdd(props) {
    const { assistiveText , ...otherProps } = props;
    const isMobile = useMedia({
        maxWidth: tokens.breakpointMedium
    });
    return isMobile ? jsx(IconButton, {
        isLoading: otherProps.isLoading,
        name: "plus",
        assistiveText: assistiveText,
        onClick: otherProps.onClick,
        testHook: otherProps.testHook,
        strokeWidth: "2"
    }, void 0) : jsx(Button, Object.assign({
        variant: "bordered"
    }, otherProps), void 0);
}
function ButtonAddLoader(props) {
    const isMobile = useMedia({
        maxWidth: tokens.breakpointMedium
    });
    return isMobile ? jsx(Skeleton, {
        borderRadius: "borderRadiusRoot",
        size: 40
    }, void 0) : jsx(ButtonLoader, Object.assign({}, props), void 0);
}
ButtonAdd.Loader = ButtonAddLoader;

export { ButtonAddLoader, ButtonAdd as default };
