// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/ThemeProvider/ThemeProvider.js
import { jsx } from 'react/jsx-runtime';
import { ThemeProvider as ThemeProvider$1 } from 'styled-components';
import { themeBreakpoints } from '../../../constants/tokens/breakpoints.js';
import { borderRadii } from '../../../constants/tokens/borderRadii.js';
import { borderWidths } from '../../../constants/tokens/borderWidths.js';
import { boxShadows } from '../../../constants/tokens/boxShadows.js';
import { fontFamilies } from '../../../constants/tokens/fontFamilies.js';
import { fontSizes } from '../../../constants/tokens/fontSizes.js';
import { fontWeights } from '../../../constants/tokens/fontWeights.js';
import { letterSpacings } from '../../../constants/tokens/letterSpacings.js';
import { lineHeights } from '../../../constants/tokens/lineHeights.js';
import { colors } from '../../../constants/tokens/colors.js';
import { sizings } from '../../../constants/tokens/sizings.js';
import { sizingsAvatar } from '../../../constants/tokens/sizingsAvatar.js';
import { sizingsIcon } from '../../../constants/tokens/sizingsIcon.js';
import { spacing } from '../../../constants/tokens/spacing.js';
import { transitions } from '../../../constants/tokens/transitions.js';
import { zIndexes } from '../../../constants/tokens/zIndexes.js';

const theme = {
    borderWidths,
    breakpoints: themeBreakpoints,
    colors,
    fonts: fontFamilies,
    fontSizes,
    fontWeights,
    letterSpacings,
    lineHeights: {
        ...lineHeights,
        ...sizingsAvatar,
        ...sizingsIcon
    },
    radii: borderRadii,
    shadows: boxShadows,
    sizes: sizings,
    space: spacing,
    transitions,
    zIndices: zIndexes
};
function ThemeProvider(props) {
    const { children  } = props;
    return jsx(ThemeProvider$1, Object.assign({
        theme: theme
    }, {
        children: children
    }), void 0);
}

export { ThemeProvider as default };
