// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/Illustration/Illustration.js
import { jsx } from 'react/jsx-runtime';
import cx from 'classnames';
import invariant from 'invariant';
import keyMirror from 'keymirror';
import Skeleton from '../Skeleton/Skeleton.js';
import svgs from './svgs/index.js';
import styles from './Illustration.css.js';

const defaultCategory = 'main';
keyMirror({
    brandColor: null
});
function Illustration(props) {
    const { name , size , variant  } = props;
    let category = name.split('-')[0];
    invariant(!!styles[category], `Illustration: no category style defined for name '${name}'. Defaulting to '${defaultCategory}'.`);
    if (!styles[category]) {
        category = defaultCategory;
    }
    const Component = svgs[name];
    return jsx(Component, {
        "aria-hidden": true,
        className: cx(styles.root, styles[category], {
            [String(size && styles[size])]: size,
            [String(variant && styles[variant])]: variant
        })
    }, void 0);
}
function IllustrationLoader(props) {
    return jsx(Skeleton, Object.assign({}, props, {
        borderRadius: "borderRadiusCircle",
        size: {
            base: 100,
            tablet: 250
        }
    }), void 0);
}
Illustration.Loader = IllustrationLoader;

export { Illustration as default };
