// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/QuizHome/QuizHome.js
import { jsx, jsxs } from 'react/jsx-runtime';
import Button, { ButtonLoader } from '../Button/Button.js';
import ButtonGroup from '../ButtonGroup/ButtonGroup.js';
import Heading from '../Heading/Heading.js';
import Illustration from '../Illustration/Illustration.js';
import Text, { TextLoader } from '../Text/Text.js';
import VerticalSpacing from '../VerticalSpacing/VerticalSpacing.js';
import styles from './QuizHome.css.js';

function useSkeleton(param) {
    let { actions , labels , button , illustration , heading , metadata , subheading ,  } = param;
    return jsxs("div", Object.assign({
        className: styles.root
    }, {
        children: [
            jsx(VerticalSpacing, Object.assign({
                size: "large3X"
            }, {
                children: illustration
            }), void 0),
            jsxs(VerticalSpacing, Object.assign({
                size: "large1X"
            }, {
                children: [
                    heading,
                    labels && jsx(VerticalSpacing, Object.assign({
                        size: "small2X",
                        isUpwards: true
                    }, {
                        children: labels
                    }), void 0),
                    subheading && jsx(VerticalSpacing, Object.assign({
                        size: "small1X",
                        isUpwards: true
                    }, {
                        children: subheading
                    }), void 0)
                ]
            }), void 0),
            jsx(VerticalSpacing, Object.assign({
                size: "large3X"
            }, {
                children: metadata
            }), void 0),
            actions && jsx(VerticalSpacing, Object.assign({
                size: "large4X"
            }, {
                children: actions
            }), void 0),
            jsx(VerticalSpacing, {
                children: button
            }, void 0)
        ]
    }), void 0);
}
function Loader() {
    return jsx("div", Object.assign({
        className: styles.loader
    }, {
        children: useSkeleton({
            illustration: jsx(Illustration.Loader, {
                marginInline: "auto"
            }, void 0),
            heading: jsx(TextLoader, {
                width: "80%",
                marginInline: "auto"
            }, void 0),
            labels: null,
            subheading: jsx(TextLoader, {
                width: "60%",
                marginInline: "auto"
            }, void 0),
            metadata: jsx(TextLoader, {
                width: "40%",
                marginInline: "auto"
            }, void 0),
            actions: jsx(ButtonGroup.Loader, {}, void 0),
            button: jsx(ButtonLoader, {
                size: "large",
                marginInline: "auto"
            }, void 0)
        })
    }), void 0);
}
function QuizHome(props) {
    const { actions , labels , buttonProps , heading , illustrationName , metadata , subheading  } = props;
    return useSkeleton({
        illustration: jsx(Illustration, {
            name: illustrationName
        }, void 0),
        heading: jsx(Heading, Object.assign({
            rank: "1",
            size: "large"
        }, {
            children: heading
        }), void 0),
        labels,
        subheading: subheading && jsx(Text, Object.assign({
            as: "p",
            variant: "lead"
        }, {
            children: subheading
        }), void 0),
        metadata: jsx("div", Object.assign({
            className: styles.metadata
        }, {
            children: jsx(Text, Object.assign({
                variant: "bodySmall1X",
                color: "colorFiftyShadesDark"
            }, {
                children: metadata
            }), void 0)
        }), void 0),
        actions,
        button: jsx(Button, Object.assign({}, buttonProps, {
            size: "large"
        }), void 0)
    });
}QuizHome.Loader = Loader;

export { QuizHome as default };
