// @getatomi/neon@33.2.2-PROD-4806.1/dist/components/public/Panel/Panel.js
import { jsxs, jsx } from 'react/jsx-runtime';
import cx from 'classnames';
import keyMirror from 'keymirror';
import Icon from '../Icon/Icon.js';
import styles from './Panel.css.js';

const variants = keyMirror({
    general: null,
    important: null,
    info: null
});
function Panel(props) {
    const { className , variant =variants.general , children  } = props;
    const withIcon = variant === variants.important;
    return jsxs("aside", Object.assign({
        className: cx(styles.root, variant && styles[variant], className)
    }, {
        children: [
            withIcon && jsx(Icon, {
                name: "lightbulb",
                color: "colorCoreBrand",
                display: {
                    base: 'none',
                    tablet: 'unset'
                },
                className: styles.icon
            }, void 0),
            jsx("div", {
                children: children
            }, void 0)
        ]
    }), void 0);
}

export { Panel as default, variants };
